src/LinearSolvers/iterativeWrapper.jl

```diff
@@ -24,6 +24,7 @@ Example:
 
 """
 type IterativeSolver<: AbstractSolver
+   AA   # matrix being solved
```

Comment:
I think this needs some more thought and maybe discussion since there is a non-trivial interaction between the forward problem type and the linear solvers type. 

Example: When using a direct solver, the forward problem should not rebuild the PDE during sensitivity matvecs, right? On the other hand, for iterative solvers, we should still check in sensmatvecs whether the PDE is built and if not, rebuild. If you look at what happens in DivSigGrad, then this is a pretty safe implementation (meaning that I can compute sensmats even before calling `getData`). 

The cost is of course the storage of the PDE when using explicit solvers. Is that a major drawback (since the factorization are dominating that costs)? Don't know the answer to this. If it is, let's think more carefully were to put this.  Maybe we can give template arguments to the forward problems...